politeiawww/user.go

```diff
@@ -1064,8 +1064,10 @@ func (p *politeiawww) processResendVerification(rv *www.ResendVerification) (*ww
 	}
 
 	// Remove the original pubkey from the cache.
-	existingPublicKey := hex.EncodeToString(u.Identities[0].Key[:])
-	p.removeUserPubkeyAssociaton(u, existingPublicKey)
+	if len(u.Identities) > 0 {
```

Comment:
I don't really like this because its not clear why its needed from looking at it.  pi makes the assumption that a user identity always exists so this would confuse me.  Can we make if more explicit.  Something like:

```
	// Remove the original pubkey from the cache.
	// The pubkey has not been set yet for cms users.
	if p.cfg.Mode == politeiaWWWMode {
		existingPublicKey := hex.EncodeToString(u.Identities[0].Key[:])
		p.removeUserPubkeyAssociaton(u, existingPublicKey)
	}
```

In general, I'm not sure if we should be adding `if p.cfg.Mode == x` statements.  It sets a bad precedent, especially 
if additional modes are added to politeia in the future. We don't want these routes to turn into a bunch of `if p.cfg.Mode == x` statements.  We may want to make a cms specific route if the behavior does not exactly match pi behavior.  I would get @marcopeereboom to weigh in.

Reply:
Yah I'm Ok with whatever you'd like to do there.  I just figured a basic sanity check isn't too far fetched.  

I agree with the issues with overusing the mode check.

Generally, it's becoming more of an issue with some of the routes in userwww aren't necessarily applicable to cms which is the need for the mode check.  So if we can figure out a decent way to not include some of the routes (like resendverification) in cms, we wouldn't really need any checks at all.